det3pa/med3pa/tree.py

```diff
@@ -0,0 +1,191 @@
+        """
+        Recursively builds the tree representation starting from the specified node.
+
+        Args:
+            dtr (DecisionTreeRegressorModel): Trained decision tree regressor model.
+            X (DataFrame): Training data features.
+            y (Series): Training data labels.
+            node_id (int): Node ID to start building from. Defaults to 0.
+            path (Optional[List[str]]): Path to the current node. Defaults to ['*'].
+
+        Returns:
+            _TreeNode: The root node of the tree representation.
+        """
+        self.nb_nodes += 1
+        left_child = dtr.model.tree_.children_left[node_id]
+        right_child = dtr.model.tree_.children_right[node_id]
+
+        node_value = y.mean()
+        node_max = y.max()
+        node_samples_ratio = dtr.model.tree_.n_node_samples[node_id] / dtr.model.tree_.n_node_samples[0] * 100
+
+        # If we are at a leaf
+        if left_child == -1:
+            curr_node = _TreeNode(value=node_value, value_max=node_max, samples_ratio=node_samples_ratio,
+                                  node_id=self.nb_nodes, path=path)
+            return curr_node
+
+        node_thresh = dtr.model.tree_.threshold[node_id]
+        node_feature_id = dtr.model.tree_.feature[node_id]
+        node_feature = self.features[node_feature_id]
+
+        curr_path = list(path)  # Copy the current path to avoid modifying the original list
+        curr_node = _TreeNode(value=node_value, value_max=node_max, samples_ratio=node_samples_ratio,
+                              threshold=node_thresh, feature=node_feature, feature_id=node_feature_id,
+                              node_id=self.nb_nodes, path=curr_path)
+
+        # Update paths for child nodes
+        left_path = curr_path + [f"{node_feature} <= {node_thresh}"]
+        right_path = curr_path + [f"{node_feature} > {node_thresh}"]
+
+        curr_node.c_left = self.build_tree(dtr, X=X.loc[X[node_feature] <= node_thresh], y=y[X[node_feature] <= node_thresh],
+                                           node_id=left_child, path=left_path)
+        curr_node.c_right = self.build_tree(dtr, X=X.loc[X[node_feature] > node_thresh], y=y[X[node_feature] > node_thresh],
+                                            node_id=right_child, path=right_path)
+
+        return curr_node
+
+    def get_all_profiles(self, min_ca: float = 0, min_samples_ratio: float = 0) -> list:
+        """
+        Retrieves all profiles from the tree that meet the minimum criteria for value and sample ratio.
+
+        Args:
+            min_ca (float): Minimum value threshold for profiles. Defaults to 0.
+            min_samples_ratio (float): Minimum sample ratio threshold for profiles. Defaults to 0.
+
+        Returns:
+            List[Profile]: A list of Profile instances meeting the specified criteria.
+        """
+        if self.head is None:
+            raise ValueError("Tree has not been built yet.")
```

Comment:
One liner with assert:
`Assert self.head is not None, "Tree has not been built yet."`